habitTrackerApp.py

```diff
@@ -275,45 +277,76 @@ class MyBoxLayout(BoxLayout):
+            "red": [(1,166/255,136/255,1),'Sing Song'],
+            "pink": [(1,174/255,236/255,1),'Sit Ups'],
+            "orange": [(1,204/255,139/255,1),'Sleeping'],
             "5": [(1,0,1,1),'Jumping'],
             "white": [(1,1,1,1),'Stretching'],
             "gridBlue": [(0.082,0.298,1,0.6),'Eat Vegetables']
             }
     buttonList = []
     TheTodayButton = ''
-    currentSideButton = ObjectProperty()
+    currentSideButton = ''
 
     def __init__(self, **kwargs):
         super(MyBoxLayout, self).__init__(**kwargs)
         self.r_smash()
         
     def r_smash(self):
-        def changeColor(self):
-            MyBoxLayout.chosenColor = self.background_color
-            print("This is chosenColor inside the changeColor" + str(MyBoxLayout.chosenColor))
-        rightLayout = BoxLayout(orientation = 'vertical', size_hint = (0.2,1))
-
-        for key, values in self.colorDict.items():
-            btn1 = SideButton()
-            btn1.genID = values[1]
-            btn1.sideButtonColor = values[0]
-            btn1.originalColor = values[0]
-            rightLayout.add_widget(btn1)
-            #Bookmark 8 Feb 2022. Work out why currentSideButton is apparently empty when the app is initialised
-            if key == "red":
-                self.currentSideButton = btn1
-                print(self.currentSideButton)
+
+        rightLayout = GridLayout(size_hint = (0.2,1), rows = 7)
+
+        with rightLayout.canvas.after:
+            Color(0, 0, 0, 1)
+            self.rect = Line(width = 2, rectangle = (self.x, self.y, self.width, self.height))
+
+        def _update_rect(instance, value):
+            self.rect.rectangle = (instance.x, instance.y, instance.width, instance.height)
+
+        rightLayout.bind(size=_update_rect)
+        
+        btn1 = SideButton(size = ('200dp','100dp'), size_hint = (1, None))
+        btn1.genID = 'blue'
+        btn1.sideButtonColor = self.colorDict.get('blue')[0]
+        btn1.originalColor = self.colorDict.get('blue')[0]
+        self.currentSideButton = btn1
+
+        btn2 = SideButton(size = ('200dp','100dp'), size_hint = (1, None))
+        btn2.genID = 'green'
+        btn2.sideButtonColor = self.colorDict.get('green')[0]
+        btn2.originalColor = self.colorDict.get('green')[0]
+
+        btn3 = SideButton(size = ('200dp','100dp'), size_hint = (1, None))
+        btn3.genID = 'red'
+        btn3.sideButtonColor = self.colorDict.get('red')[0]
+        btn3.originalColor = self.colorDict.get('red')[0]
+
+        btn4 = SideButton(size = ('200dp','100dp'), size_hint = (1, None))
+        btn4.genID = 'pink'
+        btn4.sideButtonColor = self.colorDict.get('pink')[0]
+        btn4.originalColor = self.colorDict.get('pink')[0]
+
+        btn5 = SideButton(size = ('200dp','100dp'), size_hint = (1, None))
+        btn5.genID = 'orange'
+        btn5.sideButtonColor = self.colorDict.get('orange')[0]
+        btn5.originalColor = self.colorDict.get('orange')[0]
```

Comment:
The only difference between all these button is the color which you can just pass as an argument to the constructor of SideButton and you would only have 4 line : 

```suggestion
        btn1 = SideButton(size = ('200dp','100dp'), size_hint = (1, None), color='blue')
        self.currentSideButton = btn1

        btn2 = SideButton(size = ('200dp','100dp'), size_hint = (1, None), color='green')
        btn3 = SideButton(size = ('200dp','100dp'), size_hint = (1, None), color='red')
        btn4 = SideButton(size = ('200dp','100dp'), size_hint = (1, None), color='pink')
        btn5 = SideButton(size = ('200dp','100dp'), size_hint = (1, None), color='orange')

```